viewing_party/party.py

```diff
@@ -1,23 +1,179 @@
+from collections import Counter
+from shutil import move
+
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {
+        "title":title,
+        "genre":genre,
+        "rating":rating
+    }
+
+    for val in movie_dict.values():
+        if not val:
+            return None
+
+    return movie_dict
+
+def add_to_watched(user, movie):
+
+    for key in user:
+        user[key].append(movie)
+    
+    return user
+
+def add_to_watchlist(user, movie):
+
+    for key in user:
+        user[key].append(movie)
+    
+    return user
+
+def watch_movie(user_data, movie):
+    
+    filtered_user_data = dict(user_data)
+
+    for elem in user_data["watchlist"]:
+        watched_movie = elem["title"]
+        if movie == watched_movie:
+            filtered_user_data["watched"].append(elem)
+            filtered_user_data["watchlist"].remove(elem)

```

Comment:
Here I might suggest that you actually add a break in your if block.

Note that, while we might only expect there to be one movie with a particular title in the watchlist (might not be a good assumption, but for this simplified data, it might be ok), this implementation is fine.

As soon as we make a change (like removing from watchlist) we want to stop iterating, so we use break. By adding break after line 41, the loop iterates the watchlist, and when the desired movie is found, adds it to the watched list and removes it from the watchlist. If we returned immediately at this point, everything would be fine!

Without the break statement, we might be in a scenario where title actually matches to more than one movie as it loops (without the break) and we'll see some weirdness.